### ISSUE TITLE
Using apt-offline behind a proxy requiring a username and password

### ISSUE THREAD

**Issue by styler96** [rickysarraf/apt-offline#76]
Whenever I try to use apt-offline under a proxy that requires authentication, I tried the following command:
`apt-offline get apt-offline.sig --bundle apt-offline.zip`
which obviously didn't work because it was behind a proxy. So then I tried:
`apt-offline get apt-offline.sig --bundle apt-offline.zip --proxy-host proxy.myproxy.com --proxy-port 8080`
This said `ERROR: Proxy Authentication Required`. So I tried the same syntax as I use whenever I want to use a username and a password in a URL:
`apt-offline get apt-offline.sig --bundle apt-offline.zip --proxy-host username:password@proxy.myproxy.com --proxy-port 8080`
which still gave the same `ERROR: Proxy Authentication Required`.

So how can I get apt-offline to work with a proxy needing a username and password?

**Comment by styler96** [comment]
I wasn't trying it with the latest git version; I have now and can confirm the issue is still in the latest commit.

**Comment by rickysarraf** [comment]
Yes. That is because Proxy Authentication is something not supported yet. There's a similar PR peding, #69 , for https auth. Maybe that's something that'll be needed to extend for Proxy auth too.

But I won't be able to do it any time soon. Even that PR has been too long pending. I'm just very occupied with personal life priorities. If you have a patch to contribute, please do so. Otherwise, this'll have to wait a couple of months. Thanks.

**Comment by jeffedelsten** [comment]
Is anyone reading this able to provide a hint towards implementing proxy authentication?

**Comment by jeffedelsten** [comment]
FYI, http_proxy and https_proxy environment variables DO appear to be honoured, so one can always set the authentication credentials here instead.
